### DISCUSSION TITLE
"Permission denied (os error 13)" by running Function on Docker local-dev

### DISCUSSION THREAD

**Discussion by Pictor13** [supabase/supabase#14996] [Questions]
On local-development there seems to be an issue with filesystem permissions over the `deno` Docker container.

When trying to access an Edge Function file, Deno fails with `error: Permission denied (os error 13)`.  
I `docker-exec -it $deno_container bash` into `docker ps | grep deno` and can see that:
- `id` returns the user `deno`, having UID & GID equal to `1993`
- `ls -la /home/deno/functions/my-function/index.ts` shows onwer UID and group GID as equal to `1000`
  - uid/gid `1000` is usually corresponding to the `docker` user; when it appears in a container usually is the sign of some _volume mount_ of some sort.

Is this the case?  
I saw the `/app/Dockerfile` simply _copies_ the files, from localhost to the container, but I don't see why they should have this mismatching file permissions. 

Can somebody check in their working docker local-dev if the 2 commands above return the same results as me?

_At the moment I am unable to test functions locally._  
_What could be causing the error?_
_What can I try to restore the correct functionality?_
_Could this be a bug?_

------

#### More info

I run:
- `supabase start`
- `supabase functions serve my-function` --debug

it outputs:

> Download https://esm.sh/v125/@supabase/storage-js@2.5.1/dist/module/packages/StorageBucketApi.d.ts
Download https://esm.sh/v125/@supabase/storage-js@2.5.1/dist/module/packages/StorageFileApi.d.ts
Download https://deno.land/std@0.150.0/media_types/mod.ts
Download https://esm.sh/v125/@supabase/realtime-js@2.7.2/deno/dist/module/lib/version.js
Download https://esm.sh/v125/node.ns.d.ts
Download https://deno.land/std@0.150.0/media_types/_util.ts
Download https://deno.land/std@0.150.0/media_types/vendor/mime-db.v1.52.0.ts
2023/06/13 03:17:16 Sent Header: Host [docker]
2023/06/13 03:17:16 Sent Header: User-Agent [Go-http-client/1.1]
2023/06/13 03:17:16 Send Done
2023/06/13 03:17:16 Recv First Byte
Serving supabase/functions/my-function
2023/06/13 03:17:16 Sent Header: Host [docker]
2023/06/13 03:17:16 Sent Header: User-Agent [Go-http-client/1.1]
2023/06/13 03:17:16 Sent Header: Content-Length [805]
2023/06/13 03:17:16 Sent Header: Content-Type [application/json]
2023/06/13 03:17:16 Send Done
2023/06/13 03:17:16 Recv First Byte
Watcher Process started.
Check file:///home/deno/functions/my-function/index.ts
error: Permission denied (os error 13)
Watcher Process finished. Restarting on file change...

#### Additionally

- I also tried `supabase stop` and `start` again to see if it would reset the image & containers, but nothing changed
- I tried to `chmod 777` the `index.ts` file inside the container but user `deno` has no rights to do so and `sudo` is not available.

**Comment by laktek** [comment]
Closing the discussion in favor of this issue https://github.com/supabase/supabase/issues/15040
